sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/concatenate_data/cloud_collector.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "combine_cloud_handler.hpp"
+
+#include <memory>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+
+namespace autoware::pointcloud_preprocessor
+{
+
+class PointCloudConcatenateDataSynchronizerComponent;
+class CombineCloudHandler;
+
+enum class CollectorStrategyType { Naive, Advanced };
+
+struct CollectorInfo
+{
+  double timestamp{0.0};
+  double noise_window{0.0};
+  CollectorStrategyType strategy_type{CollectorStrategyType::Naive};
```

Comment:
This is not polymorphic - the problem discussed in my previous comment has just been moved into this struct:

`noise_window` is just a dead field when the strategy is Naive.
Please make sure all issues in the original comment are addressed.

Reply:
I understand! I was considering the cost of using `dynamic_pointer_cast`. 
commit https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/fd7db92821b09e31edda9144aaf558f58e92be6d fixed the problem.
Thanks for the review!!!!